sympy/polys/agca/ideals.py

```diff
@@ -326,7 +326,7 @@ def gens(self):
         >>> from sympy import QQ
         >>> from sympy.abc import x, y
         >>> list(QQ.old_poly_ring(x, y).ideal(x, y, x**2 + y).gens)
-        [DMP([[1], []], QQ), DMP([[1, 0]], QQ), DMP([[1], [], [1, 0]], QQ)]
+        [DMP_Python([[1], []], QQ), DMP_Python([[1, 0]], QQ), DMP_Python([[1], [], [1, 0]], QQ)]
```

Comment:
I thought it was important to see the difference between `DMP_Python` and `DUP_Flint` so I made them print differently. At the Poly level it all looks the same but at this level I think that it is important to see what representation is being used.

Something like this ideal could be changed to print better but it should be done by using some other type rather than DMP.

Reply:
I like having the verbose printing of the class name and also the printing of lists of lists. That makes it pretty clear that this isn't an object you should be working with directly.